### DISCUSSION TITLE
Config defaults

### DISCUSSION THREAD

**Discussion by BEagle1984** [edenhill/librdkafka#3299] [Q&A] [answered]
Just to be sure... I noticed that the default configuration values documented for librdkafka (https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md) are very different than the ones in the Kafka documentation (https://kafka.apache.org/documentation).

For example here the default `batch.size` seems to be `1000000`, while Kafka describes it to be `16384` (https://kafka.apache.org/documentation/#producerconfigs_batch.size).

Are the defaults in the documentation of librdkafka correct? Are the Kafka defaults somehow "overridden" (batch.size is a client-side thing at the end of the day)?

**Comment by BEagle1984** [comment]
cc @edenhill

**Comment by edenhill** [comment, marked as answer]
While we try to maintain the same default behaviour as the Java clients, there are some instances where we can't (because of historical (librdkafka predates the Java clients and has a strict backwards compatible modus) or implementation reasons) or won't (because we disagree).

Our defaults are balanced between latency and throughput and should work well for the majority of applications, and it is generally not recommended to tweak things like batch sizes and timeouts unless really necessary.

The properties that are marked with high importance are the ones an application developer should consider.

**Reply by BEagle1984** [reply]
That's perfectly fine (and for my limited understanding of the matter I find your defaults better). I just wanted to be sure that your documentation is correct and the one to be looked at when working with librdkafka.

Thank you!

**Reply by edenhill** [reply]
Yes, CONFIGURATION.md is generated from the configuration code, so it is always up to date.
